addon/orm/model.js

```diff
@@ -271,7 +271,7 @@ class Model {
     let associations = this.schema.associationsFor(this.modelName);
     let matchingExplicitInverses = Object.keys(associations).filter(key => {
       let candidate = associations[key];
-      let modelMatches = association.modelName === candidate.ownerModelName;
+      let modelMatches = association.modelName === candidate.ownerModelName && association.ownerModelName === candidate.modelName;
```

Comment:
Miraculously, the existing code is simply wrong. Not sure how it managed to survive this long.

We're making sure there's only one association on `user` that matches `post.authors`, so we want to check all the associations on `user` (the candidates) whose _modelName_ matches the given association's _ownerModelName_ (and whose inverse key matches).

The code should be

```js
let modelMatches = association.ownerModelName === candidate.modelName;
```

Reply:
Cool, it felt like that might be true to me, but I was completely unable to keep all the association and model names straight. I must have stared at my debugger for 15 minutes just looking.  
Anyway, I just pushed an amended commit with only the correct check.

Reply:
haha I know... very confusing part of the codebase. Always open to improvements if you have any ideas!

Thanks!